lib/puppet/type/acl/ace.rb

```diff
@@ -29,14 +29,17 @@ def initialize(permission_hash, provider = nil)
       self.id = permission_hash['id']
       @mask = permission_hash['mask']
       self.rights = permission_hash['rights']
-      self.type = permission_hash['type']
+      self.perm_type = permission_hash['perm_type']
+      if permission_hash['type']
+        self.perm_type = permission_hash['type']
+      end
```

Comment:
You can't b/c perm_type has a default value and therefore will always have a value.